src/background/messenger.ts

```diff
@@ -106,6 +129,8 @@ export default class Messenger {
     }
 
     reportChanges(data: ExtensionData) {
-        this.reporters.forEach((report) => report(data));
+        if (this.changeListenerCount > 0 || !isChromium) {
```

Comment:
On Chromium (determined as `!isFirefox`) connector sends `subscribe-to-changes` and `unsubscribe-from-chages` which inform messenger to increment/decrement `this.changeListenerCount`. So when there is at least one active subscriber, `changeListenerCount > 0` is `true`. On Firefox (determined by `isFirefox`) we don't use `this.changeListenerCount` and just broadcast change notifications unconditionally.

This optimization is needed on Chromium to rule out unnecessary change notification messages which could cause unnecessary onMessage event handler work. On FIrefox, I'm hesitant to use this optimization because Firefox is very racy.